From fb1dec7e2cbb2eae14536d8d68db25d3f9384ac1 Mon Sep 17 00:00:00 2001
From: aymenim <aymen_ibrahim@hotmail.com>
Date: Tue, 15 Mar 2016 23:56:08 +0300
Subject: [PATCH] cython: fixed notification_proxy callback gil lock

cython notification proxy bug segmentation fault when callback called
without gil
---
 cython/notification_proxy.pxi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cython/notification_proxy.pxi b/cython/notification_proxy.pxi
index 4ffbf0742..203c7835d 100644
--- a/cython/notification_proxy.pxi
+++ b/cython/notification_proxy.pxi
@@ -70,7 +70,7 @@ NP_ITDBPREP_DID_END = C_NP_ITDBPREP_DID_END
 NP_LANGUAGE_CHANGED = C_NP_LANGUAGE_CHANGED
 NP_ADDRESS_BOOK_PREF_CHANGED = C_NP_ADDRESS_BOOK_PREF_CHANGED
 
-cdef void np_notify_cb(const_char_ptr notification, void *py_callback):
+cdef void np_notify_cb(const_char_ptr notification, void *py_callback) with gil:
     (<object>py_callback)(notification)
 
 cdef class NotificationProxyError(BaseError):
